caql/CaQLParser.g4

```diff
@@ -0,0 +1,87 @@
+parser grammar CaQLParser;
+
+options { tokenVocab = CaQLLexer; }
+
+expression: vectorOperation;
+
+// Binary operations are ordered by precedence
+
+// Unary operations have the same precedence as multiplications
+
+vectorOperation
+    : <assoc=right> vectorOperation powOp vectorOperation # assocRight
+    | <assoc=right> vectorOperation subqueryOp  # assocRight
+    | unaryOp vectorOperation   # unary
+    | vectorOperation multOp vectorOperation #mult
+    | vectorOperation addOp vectorOperation  #add
+    | vectorOperation compareOp vectorOperation # compare
+    | vectorOperation andUnlessOp vectorOperation # andUnless
+    | vectorOperation orOp vectorOperation #or
+    | vectorOperation vectorMatchOp vectorOperation #match
+
+    | FUNCTION LEFT_BRACE vectorOperation (multMetrics)*  RIGHT_BRACE # opMethod
+    | LEFT_PAREN vectorOperation RIGHT_PAREN #paren
+    | FUNCTION LEFT_PAREN  metricsName=STRING COMMA  condition=STRING RIGHT_PAREN (metricsAggregation)* #fuction
+    | FUNCTION LEFT_PAREN  metricsName=STRING COMMA  condition=STRING COMMA  LIMIT size=NUMBER RIGHT_PAREN (metricsAggregation)* #fuction
+    | vectorOperation metricsAggregation # pip
+    | NUMBER #number
+    ;
+// Operators
+
+
+unaryOp:        (ADD | SUB);
+powOp:          POW grouping?;
+multOp:         (MULT | DIV | MOD) grouping?;
+addOp:          (ADD | SUB) grouping?;
+compareOp:      (DEQ | NE | GT | LT | GE | LE) BOOL? grouping?;
+andUnlessOp:    (AND | UNLESS) grouping?;
+orOp:           OR grouping?;
+vectorMatchOp:  (ON | UNLESS) grouping?;
+subqueryOp:     SUBQUERY_RANGE offsetOp?;
+offsetOp:       OFFSET DURATION;
+
+// Functions
+
+parameter:     literal | vectorOperation;
+parameterList: LEFT_PAREN (parameter (COMMA parameter)*)? RIGHT_PAREN;
+metricsAggregation: Line AGGREGATION_OPERATOR parameterList;
+multMetrics: COMMA vectorOperation;
+//metricsCondition:  oop=(AND|OR) LEFT_PAREN METRIC_NAME METRIC_NAME (metricsMult)* RIGHT_PAREN  ;
+//metricsMult: COMMA METRIC_NAME  METRIC_NAME;

```

Comment:
Please remove commented out code and create an issue if necessary.

Reply:
removed